### PR TITLE
[18.01] Client related readme/makefile clarifications

### PR DESCRIPTION
**Pull request by dannon**
Made the client README slightly easier to find.  Updated the Makefile comments to reflect the sourcemap compromise.  Minor tweaks to client README.

Also did a text reflow in CONTRIBUTING to standardize line breaks.

xref https://github.com/galaxyproject/galaxy/issues/5483

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -33,23 +32,23 @@ opened using [this form](https://github.com/galaxyproject/galaxy/issues/new).
...
+  good places to start.  More learning resources are listed at
+  https://help.github.com/articles/good-resources-for-learning-git-and-github/
 
 * Make sure you have a free [GitHub](https://github.com/) account.
 
 * Fork the [galaxy repository](https://github.com/galaxyproject/galaxy) on
-  GitHub to make your changes.
-  (While many Galaxy instances track
+  GitHub to make your changes.  (While many Galaxy instances track
   [galaxy-dist](https://bitbucket.org/galaxy/galaxy-dist), active development
```
Comment (dannon) [inline CONTRIBUTING.md]: Should we drop this bit about bitbucket/galaxy/galaxy-dist at this point?

Comment (nsoranzo) [reply CONTRIBUTING.md]: Yes, please!

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,30 +1,29 @@
...
+Galaxy Community Conference presentation on the topic
 ([video](https://vimeo.com/channels/581875/73486255),
-[presentation](https://depot.galaxyproject.org/hub/attachments/documents/presentations/gcc2013/BakerContribute.pdf)). For
-information on contributing more broadly to the Galaxy ecosystem and a
+[presentation](https://depot.galaxyproject.org/hub/attachments/documents/presentations/gcc2013/BakerContribute.pdf)).
+For information on contributing more broadly to the Galaxy ecosystem and a
 deeper discussion of some of these points - please see the
-[Develop](https://galaxyproject.org/develop) section of the
-[Galaxy Community Hub](https://galaxyproject.org).
+[Develop](https://galaxyproject.org/develop) section of the [Galaxy Community
```
Comment (nsoranzo) [inline CONTRIBUTING.md:11]: I think we should someday merge the content of https://galaxyproject.org/develop/best-practices/ in `CONTRIBUTING.md` (not asking you to do it, of course!)

Comment (dannon) [reply CONTRIBUTING.md:11]: @nsoranzo Yeah, I was looking at that, too.  This bit about maintaining an updated fork seems like it should be there and not here, among other things.  (but, agreed, let's do it in -dev)

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,30 +1,29 @@
...
-comment on it. We also maintain a [tag](https://github.com/galaxyproject/galaxy/issues?q=is%3Aissue+is%3Aopen+label%3Afriendliness%2Ffriendly) on Github for
-smaller issues we believe would make the best entry points for new
-developers.
-Galaxy developers are generally available via
-[IRC](https://galaxyproject.org/get-involved), [Gitter](https://gitter.im/galaxyproject/Lobby)
-and on the [development mailing list](http://dev.list.galaxyproject.org).
+If you have an idea for a feature to add or an approach for a bugfix, it is
+best to communicate with Galaxy developers early. The most common venues for
+this are [GitHub issues](https://github.com/galaxyproject/galaxy/issues).
+Browse through existing GitHub issues and Trello cards and if one seems
```
Comment (nsoranzo) [inline CONTRIBUTING.md]: Time to also drop the references to Trello?

Comment (dannon) [reply CONTRIBUTING.md]: @nsoranzo Sure, one sec.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -14,21 +14,23 @@ Hub](https://galaxyproject.org).
...
-this are [GitHub issues](https://github.com/galaxyproject/galaxy/issues).
-Browse through existing GitHub issues and Trello cards and if one seems
-related, comment on it. We also maintain a
-[tag](https://github.com/galaxyproject/galaxy/issues?q=is%3Aissue+is%3Aopen+label%3Afriendliness%2Ffriendly)
-on Github for smaller issues we believe would make the best entry points for
-new developers.  Galaxy developers are generally available via
-[IRC](https://galaxyproject.org/get-involved),
+best to communicate with Galaxy developers early. The primary venue for this is
+the [GitHub issue tracker](https://github.com/galaxyproject/galaxy/issues).
+Browse through existing GitHub issues and and if one seems related, comment on
```
Comment (nsoranzo) [inline CONTRIBUTING.md]: and and

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -14,21 +14,23 @@ Hub](https://galaxyproject.org).
...
+it.  For more direct communication, Galaxy developers are generally available
+via [IRC](https://galaxyproject.org/get-involved),
 [Gitter](https://gitter.im/galaxyproject/Lobby) and on the [development mailing
 list](http://dev.list.galaxyproject.org).
 
+If you're looking to help but aren't sure where to start, we also maintain a
+[tag](https://github.com/galaxyproject/galaxy/issues?q=is%3Aissue+is%3Aopen+label%3Afriendliness%2Ffriendly)
+on Github for smaller issues we believe would make the best entry points for
```
Comment (nsoranzo) [inline CONTRIBUTING.md]: s/Github/GitHub/

**Review by nsoranzo — APPROVED** [review]

**Pull request merged**